crypto/hpke/hpke_util.c

```diff
@@ -82,18 +70,27 @@ int ossl_hpke_kdf_expand(EVP_KDF_CTX *kctx,
 int ossl_hpke_labeled_extract(EVP_KDF_CTX *kctx,
                               unsigned char *prk, size_t prklen,
                               const unsigned char *salt, size_t saltlen,
+                              const char *protocol_label,
                               const unsigned char *suiteid, size_t suiteidlen,
                               const char *label,
                               const unsigned char *ikm, size_t ikmlen)
 {
     int ret = 0;
     size_t labeled_ikmlen = 0;
-    unsigned char labeled_ikm[LABELED_EXTRACT_SIZE];
+    unsigned char *labeled_ikm = NULL;
     WPACKET pkt;
 
+    labeled_ikmlen = strlen(LABEL_HPKEV1) + strlen(protocol_label)
+                     + suiteidlen + strlen(label) + ikmlen;
+    labeled_ikm = OPENSSL_malloc(labeled_ikmlen);
+    if (labeled_ikm == NULL) {
+        ERR_raise(ERR_LIB_PROV, ERR_R_MALLOC_FAILURE);
```

Comment:
This was the biggest outstanding thing that needed doing here..
Don't doing the raise for malloc failure, since this is being changed in another PR so that the malloc does it. So yopu can remove the {} also..  
You should probably just return here since it would crash on the pkt cleanup.

Reply:
ack, doing that